### PR TITLE
windowchef: init at 0.5.0

### PR DESCRIPTION
**Pull request by bhougland18**
###### Motivation for this change
Windowchef is a stacking window manager similar to 2bwm, but more modular in its approach.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/windowchef/default.nix
Comment (cole-h) [inline pkgs/applications/window-managers/windowchef/default.nix:7]:
```suggestion
  version = "0.5.0";
```

We prefer not to use letters in versions.

File: pkgs/applications/window-managers/windowchef/default.nix
Comment (cole-h) [inline pkgs/applications/window-managers/windowchef/default.nix:12]:
```suggestion
    rev    = "v${version}";
```

As a corollary to the above.

File: pkgs/applications/window-managers/windowchef/default.nix
Comment (cole-h) [inline pkgs/applications/window-managers/windowchef/default.nix:22]:
```suggestion
    homepage = "https://github.com/tudurom/windowchef";
```

**Review by cole-h — CHANGES_REQUESTED** [review]
A few changes for after you fix the merge conflict. Also, it is preferable that you add yourself as a maintainer in a separate commit, `maintainers: add bhougland18`.

File: pkgs/applications/window-managers/windowchef/default.nix
Comment (bhougland18) [reply pkgs/applications/window-managers/windowchef/default.nix:7]: This is actually a unique circumstance because while I was creating this package I was added as a maintainer on commit #84284.  However, when I looked it was bhougland instead of bhougland18, so i acually removed myself from the maintainers on the third commit.  It was a crazy timing issue.

File: pkgs/applications/window-managers/windowchef/default.nix
Comment (bhougland18) [reply pkgs/applications/window-managers/windowchef/default.nix:12]: I initially had just the version but it doesn't find the tar.gz package (the url string include the prefix of v.  Is there a better place to put this?

Comment (cole-h) [reply pkgs/applications/window-managers/windowchef/default.nix]: Since `fetchFromGitHub` takes the tarball from the tag, this shouldn't matter. If you're dealing with a raw URL, though, you can do as I did here: `v${version}`, e.g.:

`https://github.com/ajeetdsouza/zoxide/archive/v${version}.tar.gz`

Whatever you need to not have any letters in the package version.

File: pkgs/applications/window-managers/windowchef/default.nix
Comment (cole-h) [inline pkgs/applications/window-managers/windowchef/default.nix:23]: I think the best course of action is to change this to what is present in the current maintainers list (`bhougland`) and submit a followup PR later, renaming all instances of `bhougland` to `bhougland18`.

Or, you could just let your maintainer handle be `bhougland` and allow it to be slightly different from your GitHub handle.

Either way works.

**Review by bhougland18 — COMMENTED** [review]
windowchef: updated maintainer

File: pkgs/applications/window-managers/windowchef/default.nix
Comment (bhougland18) [reply pkgs/applications/window-managers/windowchef/default.nix]: I will just leave it as bhougland, I use that username many places.

Comment (bhougland18) [reply pkgs/applications/window-managers/windowchef/default.nix]: Should I do a separate commit for the service module similar to https://github.com/NixOS/nixpkgs/blob/release-19.09/nixos/modules/services/x11/window-managers/spectrwm.nix

Comment (cole-h) [reply pkgs/applications/window-managers/windowchef/default.nix]: If you want to add it a service for it, it should definitely be a separate commit, IMO.

Comment (bhougland18) [reply pkgs/applications/window-managers/windowchef/default.nix]: sounds good.

**Review by jonringer — COMMENTED** [review]
Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the commits.

This can be done with:
```
git reset HEAD~7
git add -- pkgs/
git commit --amend --no-edit
git push ... ... --force
```

**Comment by bhougland18** [comment]
I folllowed your steps exactly because it seems like this pull request is stuck. However, now I think there are a lot of issues.  Any help?

**Comment by bhougland18** [comment]
> Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the commits.
> 
> This can be done with:
> 
> ```
> git reset HEAD~7
> git add -- pkgs/
> git commit --amend --no-edit
> git push ... ... --force
> ```

**Comment by bhougland18** [comment]
I will try again.  However, the documentation on how to do this seems very poor.

**Comment by jonringer** [comment]
I think you had other commits on top of it, my assumption was that you were at the HEAD of the PR branch

**Comment by bhougland18** [comment]
@jonringer  no problem, sorry for the bad attitude earlier.  I think this commit could be cleaned up anyway and this helps me understand git better.

**Pull request closed without merging**
